services/wallet/collectibles/manager.go

```diff
@@ -1163,3 +1157,9 @@ func createStatusNotifier(statuses *sync.Map, feed *event.Feed) *connection.Stat
 		feed,
 	)
 }
+
+// Different providers have API keys per chain or per testnet/mainnet.
+// Proper implementation should respect that. For now, the safest solution is to use the provider ID and chain ID as the key.
+func getCircuitName(provider thirdparty.CollectibleProvider, chainID walletCommon.ChainID) string {
+	return provider.ID() + chainID.String()
```

Comment:
Should we add any delimiter in between, maybe better?
`return fmt.Sprintf("%s-%s", return provider.ID(), chainID.String())`

Reply:
I agree, just thought about it, if I need to print it somewhere, easier for eyes, not a big deal.